metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -327,6 +327,8 @@ def task_pre_step(
 
             # TODO (savin): Introduce equivalent support for Microsoft Azure and
             #               Google Cloud Platform
+            # TODO: Introduce a way to detect Cloud Provider, so unnecessary requests (and delays)
```

Comment:
Two other options to consider:
1. The information obtained from the metadata service here is also available in K8s. One option could be to get that info from K8s and avoid making this call. It will also make the implementation cloud agnostic.
2. Is it possible to make this call async? Since the info here is not directly needed for the actual task function to execute, it could be called in async.

WDYT?

Reply:
To add to this, there is a decent chance that you won't be able to call this endpoint on EKS at all, depending on node group configuration, specifically IMDS settings (IMDSv1 vs IMDSv2 and the hop limit). 

But its a known issue. When this call was introduced it was considered a "best effort" thing, longer term would indeed be better to get this info in more reliable, K8S-native way.

Reply:
This is great for now. 

Call should probably be async, it is a completely independent process. It should still need a relatively aggressive timeout though.

Ideally K8s should be responsible for the metadata of the node the step is run on to create a consistent way of working across all infrastructure setups.

Perhaps then making a flag to optionally get AWS EC2 host metadata directly.